Update to EPICS core contribution guide.

eddybl opened the pull request:
- Replaced .rst by .md file
- Replaced Launchpad with GitHub instructions
- Dropped references to old Git versions

I wasn't quite sure about the `Compability` section in regards to `vxWorks`. I guess for the older versions it is still correct, but we might want to add some specifics for EPICS 7?

Closes  #117

minijackson commented on contributing/HowToWorkWithTheEpicsRepository.md:
```diff
@@ -0,0 +1,182 @@
+You can push updates at any time,
+even after making a pull request.
+The updates will automatically be part of the merge request.
+Do **not** create a new merge request because of an update!
```
I know this isn't part of your changes, but I didn't understand this sentence at first. Maybe something like this:?

> Do **not** create a new Pull Request if you want to update your contribution.

minijackson reviewed (commented):
Thanks a lot! As a side note, I think it would have been easier to review if you separated your contribution into 2 commits: one for the conversion from rst -> md, and the other with the proposed changes. I had to manually compare the [ReadTheDocs PR](https://epics--118.org.readthedocs.build/en/118/contributing/HowToWorkWithTheEpicsRepository.html) build with the [current URL](https://docs.epics-controls.org/en/latest/contributing/HowToWorkWithTheEpicsRepository.html).

eddybl commented:
Thanks for the review, I added your suggestions (and replaced "merge request" with "pull request") and sorry for the inconvienience with the single commit only

minijackson commented:
No worries!

I'm not seeing your changes, however. I'm only seeing a merge commit, maybe a Git mishap?

ttkorhonen commented:
I was wondering the same...would be ready for merging if the fixes were there.

eddybl commented:
Sorry, syncing forks, pulling locally, comitting, pushing... apparently I mixed something up at one point :o

ttkorhonen commented:
No worries, I do it all the time myself...the PR looks good to me now, thank you, this is highly appreciated!
